src/hooks/useErrorHandling.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback, useRef } from 'react';
+
+export const useErrorHandling = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const errorTimerRef = useRef<NodeJS.Timeout | null>(null);
+
+  const handleSetErrorMessage = useCallback((message: string) => {
+    setErrorMessage(message);
+
+    if (errorTimerRef.current) {
+      clearTimeout(errorTimerRef.current);
+    }
+
+    errorTimerRef.current = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, []);
+
+  const handleClearErrorMessage = useCallback(() => {
+    setErrorMessage('');
+    if (errorTimerRef.current) {
+      clearTimeout(errorTimerRef.current);
```

Comment:
You never clear errorTimerRef.current

```suggestion
      clearTimeout(errorTimerRef.current);
      errorTimerRef.current = null;
```